### PR TITLE
Refactor Cuda CMake

### LINKED ISSUE #419: ITKs CMP0120 policy breaks RTK compatibility
Hi,

I was building ITK from master branch with RTK and CUDA.
I used VS 2019 (v16.8.7) and CMake v3.20.3 with Python v3.8.5 and CUDA Toolkit v10.2 under Windows 10.
This fails due to the following reasons:
```
75>------ Build started: Project: RTKPython, Configuration: Release x64 ------
75>itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaForwardProjectionImageFilter.cu.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaBackProjectionImageFilter.cu.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaConjugateGradientImageFilter.cu.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaUtilities.cu.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaRayCastBackProjectionImageFilter.cu.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaConstantVolumeSource.cu.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaDisplacedDetectorImageFilter.cu.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaFFTProjectionsConvolutionImageFilter.cu.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaSplatImageFilter.cu.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaInterpolateImageFilter.cu.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaConstantVolumeSeriesSource.cu.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWarpForwardProjectionImageFilter.cu.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWarpBackProjectionImageFilter.cu.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaCyclicDeformationImageFilter.cu.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaFDKWeightProjectionFilter.cu.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaCropImageFilter.cu.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaFDKBackProjectionImageFilter.cu.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "protected: __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::basic_streambuf<char,struct std::char_traits<char> >(void)" (??0?$basic_streambuf@DU?$char_traits@D@std@@@std@@IEAA@XZ) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: virtual __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::~basic_streambuf<char,struct std::char_traits<char> >(void)" (??1?$basic_streambuf@DU?$char_traits@D@std@@@std@@UEAA@XZ) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: int __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::sputc(char)" (?sputc@?$basic_streambuf@DU?$char_traits@D@std@@@std@@QEAAHD@Z) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: __int64 __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::sputn(char const *,__int64)" (?sputn@?$basic_streambuf@DU?$char_traits@D@std@@@std@@QEAA_JPEBD_J@Z) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "protected: char * __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::_Pninc(void)" (?_Pninc@?$basic_streambuf@DU?$char_traits@D@std@@@std@@IEAAPEADXZ) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: virtual __cdecl std::basic_ios<char,struct std::char_traits<char> >::~basic_ios<char,struct std::char_traits<char> >(void)" (??1?$basic_ios@DU?$char_traits@D@std@@@std@@UEAA@XZ) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: void __cdecl std::basic_ios<char,struct std::char_traits<char> >::setstate(int,bool)" (?setstate@?$basic_ios@DU?$char_traits@D@std@@@std@@QEAAXH_N@Z) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "protected: __cdecl std::basic_ios<char,struct std::char_traits<char> >::basic_ios<char,struct std::char_traits<char> >(void)" (??0?$basic_ios@DU?$char_traits@D@std@@@std@@IEAA@XZ) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: __cdecl std::basic_ostream<char,struct std::char_traits<char> >::basic_ostream<char,struct std::char_traits<char> >(class std::basic_streambuf<char,struct std::char_traits<char> > *,bool)" (??0?$basic_ostream@DU?$char_traits@D@std@@@std@@QEAA@PEAV?$basic_streambuf@DU?$char_traits@D@std@@@1@_N@Z) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: virtual __cdecl std::basic_ostream<char,struct std::char_traits<char> >::~basic_ostream<char,struct std::char_traits<char> >(void)" (??1?$basic_ostream@DU?$char_traits@D@std@@@std@@UEAA@XZ) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: void __cdecl std::basic_ostream<char,struct std::char_traits<char> >::_Osfx(void)" (?_Osfx@?$basic_ostream@DU?$char_traits@D@std@@@std@@QEAAXXZ) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: class std::basic_ostream<char,struct std::char_traits<char> > & __cdecl std::basic_ostream<char,struct std::char_traits<char> >::operator<<(int)" (??6?$basic_ostream@DU?$char_traits@D@std@@@std@@QEAAAEAV01@H@Z) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaFFTProjectionsConvolutionImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: class std::basic_ostream<char,struct std::char_traits<char> > & __cdecl std::basic_ostream<char,struct std::char_traits<char> >::flush(void)" (?flush@?$basic_ostream@DU?$char_traits@D@std@@@std@@QEAAAEAV12@XZ) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: class std::basic_ostream<char,struct std::char_traits<char> > & __cdecl std::basic_ostream<char,struct std::char_traits<char> >::operator<<(unsigned int)" (??6?$basic_ostream@DU?$char_traits@D@std@@@std@@QEAAAEAV01@I@Z) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: struct _Cvtvec __cdecl std::_Locinfo::_Getcvt(void)const " (?_Getcvt@_Locinfo@std@@QEBA?AU_Cvtvec@@XZ) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: char __cdecl std::basic_ios<char,struct std::char_traits<char> >::widen(char)const " (?widen@?$basic_ios@DU?$char_traits@D@std@@@std@@QEBADD@Z) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: class std::basic_ostream<char,struct std::char_traits<char> > & __cdecl std::basic_ostream<char,struct std::char_traits<char> >::operator<<(class std::basic_ostream<char,struct std::char_traits<char> > & (__cdecl*)(class std::basic_ostream<char,struct std::char_traits<char> > &))" (??6?$basic_ostream@DU?$char_traits@D@std@@@std@@QEAAAEAV01@P6AAEAV01@AEAV01@@Z@Z) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: class std::basic_ostream<char,struct std::char_traits<char> > & __cdecl std::basic_ostream<char,struct std::char_traits<char> >::put(char)" (?put@?$basic_ostream@DU?$char_traits@D@std@@@std@@QEAAAEAV12@D@Z) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: __cdecl std::locale::id::operator unsigned __int64(void)" (??Bid@locale@std@@QEAA_KXZ) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "protected: void __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::_Init(void)" (?_Init@?$basic_streambuf@DU?$char_traits@D@std@@@std@@IEAAXXZ) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "protected: virtual __int64 __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::xsgetn(char *,__int64)" (?xsgetn@?$basic_streambuf@DU?$char_traits@D@std@@@std@@MEAA_JPEAD_J@Z) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "protected: virtual __int64 __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::xsputn(char const *,__int64)" (?xsputn@?$basic_streambuf@DU?$char_traits@D@std@@@std@@MEAA_JPEBD_J@Z) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: void __cdecl std::basic_ios<char,struct std::char_traits<char> >::clear(int,bool)" (?clear@?$basic_ios@DU?$char_traits@D@std@@@std@@QEAAXH_N@Z) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: static unsigned __int64 __cdecl std::ctype<char>::_Getcat(class std::locale::facet const * *,class std::locale const *)" (?_Getcat@?$ctype@D@std@@SA_KPEAPEBVfacet@locale@2@PEBV42@@Z) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: class std::locale __cdecl std::ios_base::getloc(void)const " (?getloc@ios_base@std@@QEBA?AVlocale@2@XZ) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: char __cdecl std::ctype<char>::widen(char)const " (?widen@?$ctype@D@std@@QEBADD@Z) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "protected: char * __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::eback(void)const " (?eback@?$basic_streambuf@DU?$char_traits@D@std@@@std@@IEBAPEADXZ) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "protected: char * __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::gptr(void)const " (?gptr@?$basic_streambuf@DU?$char_traits@D@std@@@std@@IEBAPEADXZ) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "protected: char * __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::pbase(void)const " (?pbase@?$basic_streambuf@DU?$char_traits@D@std@@@std@@IEBAPEADXZ) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "protected: char * __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::pptr(void)const " (?pptr@?$basic_streambuf@DU?$char_traits@D@std@@@std@@IEBAPEADXZ) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "protected: char * __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::egptr(void)const " (?egptr@?$basic_streambuf@DU?$char_traits@D@std@@@std@@IEBAPEADXZ) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: int __cdecl std::ios_base::rdstate(void)const " (?rdstate@ios_base@std@@QEBAHXZ) already defined in itkRTK-5.3.lib(cuda_compile_1_generated_rtkCudaWeidingerForwardModelImageFilter.cu.obj)
75>libcpmt.lib(locale0.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(locale0.obj) : error LNK2005: "void __cdecl std::_Facet_Register(class std::_Facet_base *)" (?_Facet_Register@std@@YAXPEAV_Facet_base@1@@Z) already defined in msvcprt.lib(locale0_implib.obj)
75>libcpmt.lib(locale0.obj) : error LNK2005: "private: static class std::locale::_Locimp * __cdecl std::locale::_Getgloballocale(void)" (?_Getgloballocale@locale@std@@CAPEAV_Locimp@12@XZ) already defined in msvcprt.lib(MSVCP140.dll)
75>libcpmt.lib(locale0.obj) : error LNK2005: "private: static class std::locale::_Locimp * __cdecl std::locale::_Init(bool)" (?_Init@locale@std@@CAPEAV_Locimp@12@_N@Z) already defined in msvcprt.lib(MSVCP140.dll)
75>libcpmt.lib(locale0.obj) : error LNK2005: "public: static void __cdecl std::_Locinfo::_Locinfo_ctor(class std::_Locinfo *,char const *)" (?_Locinfo_ctor@_Locinfo@std@@SAXPEAV12@PEBD@Z) already defined in msvcprt.lib(MSVCP140.dll)
75>libcpmt.lib(locale0.obj) : error LNK2005: "public: static void __cdecl std::_Locinfo::_Locinfo_dtor(class std::_Locinfo *)" (?_Locinfo_dtor@_Locinfo@std@@SAXPEAV12@@Z) already defined in msvcprt.lib(MSVCP140.dll)
75>libcpmt.lib(locale0.obj) : error LNK2005: "public: static class std::locale const & __cdecl std::locale::classic(void)" (?classic@locale@std@@SAAEBV12@XZ) already defined in msvcprt.lib(MSVCP140.dll)
75>libcpmt.lib(locale.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(locale.obj) : error LNK2005: "public: static unsigned __int64 __cdecl std::codecvt<char,char,struct _Mbstatet>::_Getcat(class std::locale::facet const * *,class std::locale const *)" (?_Getcat@?$codecvt@DDU_Mbstatet@@@std@@SA_KPEAPEBVfacet@locale@2@PEBV42@@Z) already defined in msvcprt.lib(MSVCP140.dll)
75>libcpmt.lib(locale.obj) : error LNK2005: "public: static class std::locale __cdecl std::locale::global(class std::locale const &)" (?global@locale@std@@SA?AV12@AEBV12@@Z) already defined in msvcprt.lib(MSVCP140.dll)
75>libcpmt.lib(locale.obj) : error LNK2005: "public: int __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::sbumpc(void)" (?sbumpc@?$basic_streambuf@DU?$char_traits@D@std@@@std@@QEAAHXZ) already defined in msvcprt.lib(MSVCP140.dll)
75>libcpmt.lib(locale.obj) : error LNK2005: "public: int __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::sgetc(void)" (?sgetc@?$basic_streambuf@DU?$char_traits@D@std@@@std@@QEAAHXZ) already defined in msvcprt.lib(MSVCP140.dll)
75>libcpmt.lib(iosptrs.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(wlocale.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(wlocale.obj) : error LNK2005: "public: unsigned short const * __cdecl std::_Locinfo::_W_Getdays(void)const " (?_W_Getdays@_Locinfo@std@@QEBAPEBGXZ) already defined in msvcprt.lib(MSVCP140.dll)
75>libcpmt.lib(wlocale.obj) : error LNK2005: "public: unsigned short const * __cdecl std::_Locinfo::_W_Getmonths(void)const " (?_W_Getmonths@_Locinfo@std@@QEBAPEBGXZ) already defined in msvcprt.lib(MSVCP140.dll)
75>libcpmt.lib(xlocale.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(xlocale.obj) : error LNK2005: "public: unsigned short const * __cdecl std::_Locinfo::_W_Getdays(void)const " (?_W_Getdays@_Locinfo@std@@QEBAPEBGXZ) already defined in msvcprt.lib(MSVCP140.dll)
75>libcpmt.lib(xlocale.obj) : error LNK2005: "public: unsigned short const * __cdecl std::_Locinfo::_W_Getmonths(void)const " (?_W_Getmonths@_Locinfo@std@@QEBAPEBGXZ) already defined in msvcprt.lib(MSVCP140.dll)
75>libcpmt.lib(xlocale.obj) : error LNK2005: "public: int __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::sbumpc(void)" (?sbumpc@?$basic_streambuf@DU?$char_traits@D@std@@@std@@QEAAHXZ) already defined in msvcprt.lib(MSVCP140.dll)
75>libcpmt.lib(xlocale.obj) : error LNK2005: "public: int __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::sgetc(void)" (?sgetc@?$basic_streambuf@DU?$char_traits@D@std@@@std@@QEAAHXZ) already defined in msvcprt.lib(MSVCP140.dll)
75>libcpmt.lib(xstol.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(xstoul.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(xstoll.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(xstoull.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(xlock.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(xlock.obj) : error LNK2005: "public: __cdecl std::_Lockit::_Lockit(int)" (??0_Lockit@std@@QEAA@H@Z) already defined in msvcprt.lib(MSVCP140.dll)
75>libcpmt.lib(xlock.obj) : error LNK2005: "public: __cdecl std::_Lockit::~_Lockit(void)" (??1_Lockit@std@@QEAA@XZ) already defined in msvcprt.lib(MSVCP140.dll)
75>libcpmt.lib(xstrcoll.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(xdateord.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(xwctomb.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(xwctomb.obj) : error LNK2005: _Getcvt already defined in msvcprt.lib(MSVCP140.dll)
75>libcpmt.lib(xwcscoll.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(xwcsxfrm.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(xgetwctype.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(xtowlower.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(xtowupper.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(xstrxfrm.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(xmtx.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(StlCompareStringA.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(winapinls.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(StlCompareStringW.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(StlLCMapStringW.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(StlLCMapStringA.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>libcpmt.lib(winapisupp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in RTKPython.obj
75>   Creating library C:/ITK/bin-withRTKandCUDA-vs/lib/Release/RTKPython.lib and object C:/ITK/bin-withRTKandCUDA-vs/lib/Release/RTKPython.exp
75>LINK : warning LNK4098: defaultlib 'LIBCMT' conflicts with use of other libs; use /NODEFAULTLIB:library
75>C:\ITK\bin-withRTKandCUDA-vs\Wrapping\Generators\Python\itk\_RTKPython.pyd : fatal error LNK1169: one or more multiply defined symbols found
75>Done building project "RTKPython.vcxproj" -- FAILED.
```

The CMake flags I used:
```
-G "Visual Studio 16 2019"
-A x64
-DBUILD_EXAMPLES=OFF
-DBUILD_SHARED_LIBS=OFF
-DBUILD_TESTING=OFF 
-DITK_LEGACY_SILENT=ON 
-DITK_USE_GPU=ON 
-DITK_WRAP_PYTHON=ON 
-DITK_WRAP_unsigned_long_long=ON 
-DITK_WRAP_signed_long_long=ON 
-DModule_RTK=ON 
-DModule_ITKCudaCommon=ON 
-DRTK_USE_CUDA=ON 
-DCUDA_HAVE_GPU=ON 
-DRTK_BUILD_APPLICATIONS=ON
```

I compared compiler flags and linker settings with older working releases but couldn't spot a difference.

### PR DESCRIPTION
**Pull request by SimonRit**
Refactoring the CMake CUDA code to follow the evolution of the management of CUDA by CMake. It will hopefully fix #419.

### CODE REVIEW

File: cmake/FindCUDA_wrap.cmake [hunk 1]
```diff
@@ -1,80 +1,52 @@
...
-endif()
-set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS}
-     -Wno-deprecated-gpu-targets
-     -gencode arch=compute_35,code=sm_35
-     -gencode arch=compute_35,code=compute_35
-     )
+  # Allow for multiple CL.EXE to write to the same .PDB file
+  if(WIN32)
+    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /FS")
+  endif()
```
Comment (SimonRit) [inline cmake/FindCUDA_wrap.cmake]: @LucasGandel This has been moved from the main `CMakeLists.txt`. Can you confirm that it should be kept?

Comment (LucasGandel) [reply cmake/FindCUDA_wrap.cmake]: @SimonRit I just tried it with VisualStudio2019 and it seems to build fine without it. So I think you can remove it.

**Comment by SimonRit** [comment]
This PR is ready to be merged. See [here](https://cliutils.gitlab.io/modern-cmake/chapters/packages/CUDA.html) for more information on this modernization of the CMake code for CUDA.
In terms of "backward incompatibilities", I have decided to drop the ability to detect if a GPU is available, assuming that a computer with `nvcc` must have a GPU, and I also dropped the automated setting of multiple CUDA architectures since this can now be set via CMake.
It seems to me that this new version should be easier to support but please comment quickly if you have a different opinion.

**Comment by LucasGandel** [comment]
LGTM +1

**Pull request merged**